api/src/routes/settings.test.ts

```diff
@@ -122,6 +127,220 @@ describe('settingRoutes', () => {
+            id: tokenWithMissingUser,
+            ttl: 1000,
+            // Random ObjectId
+            userId: '6650ac23ccc46c0349a86dee'
+          }
+        });
+
+        await fastifyTestInstance.prisma.authToken.create({
+          data: {
+            created: new Date(Date.now() - 1000),
+            id: expiredToken,
+            ttl: 1000,
+            userId: defaultUserId
+          }
+        });
+
+        // We expect these properties to be changed by the endpoint, so they
+        // need to be set so that change can be confirmed.
+        await fastifyTestInstance.prisma.user.update({
+          where: { id: defaultUserId },
+          data: {
+            newEmail,
+            emailVerified: false,
+            emailVerifyTTL: new Date(),
+            emailAuthLinkTTL: new Date()
+          }
+        });
+      });
+
+      afterEach(async () => {
+        await fastifyTestInstance.prisma.authToken.deleteMany({
+          where: { id: { in: tokens } }
+        });
+        await fastifyTestInstance.prisma.user.update({
+          where: { id: defaultUserId },
+          data: { newEmail: null, email: defaultUserEmail, emailVerified: true }
+        });
+      });
+
+      it('should reject requests without params', async () => {
+        const resNoParams = await superGet('/confirm-email');
+
+        expect(resNoParams.headers.location).toBe(
+          'http://localhost:8000?' + formatMessage(defaultErrorMessage)
+        );
+        expect(resNoParams.status).toBe(302);
+      });
+
+      it('should reject requests which have an invalid token param', async () => {
+        const res = await superGet(
+          // token should be 64 characters long
+          `/confirm-email?email=${encodedEmail}&token=tooshort`
+        );
+
+        expect(res.headers.location).toBe(
+          'http://localhost:8000?' + formatMessage(defaultErrorMessage)
+        );
+        expect(res.status).toBe(302);
+      });
+
+      it('should reject requests which have an invalid email param', async () => {
+        const res = await superGet(
+          `/confirm-email?email=${notEmail}&token=${validToken}`
+        );
+
+        expect(res.headers.location).toBe(
+          'http://localhost:8000?' + formatMessage(defaultErrorMessage)
+        );
+        expect(res.status).toBe(302);
+      });
+
+      it('should reject requests when the auth token is not in the database', async () => {
+        const res = await superGet(
+          `/confirm-email?email=${encodedEmail}&token=${validButMissingToken}`
+        );
+
+        expect(res.headers.location).toBe(
+          'http://localhost:8000?' + formatMessage(defaultErrorMessage)
+        );
+        expect(res.status).toBe(302);
+      });
+
+      it('should reject requests when the auth token exists, but the user does not', async () => {
+        const res = await superGet(
+          `/confirm-email?email=${encodedEmail}&token=${validButMissingToken}`
```

Comment:
Shouldn't the variable then be `validButMissingUser`?

Reply:
Nope, it's the token that's missing. I'll add a comment to clarify, though, since that's not obvious.